hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/audit/AuditProcessor.java

```diff
@@ -72,7 +75,7 @@ public void process(Exchange exchange) throws Exception {
 		insert(transactionEvent);
 		
 		if (logAffectedParties) {
-			List<AffectedParty> affectedParties = createAffectedParties(v2Message, transactionId);
+			List<AffectedParty> affectedParties = createAffectedParties(v2Message, messageFlow, transactionId);
```

Comment:
May need to check for unnecessary duplicate records of the same identifier(PHN) being added. This could happen if the same identifier is repeated in request/response.